### PR TITLE
Add support for CIDR range upto /24

### PR DESCRIPTION
**Pull request by zamog**
### 1. Why is this pull request needed and what does it do?
This pull requests add support for CIDR range upto /24
### 2. Which issues (if any) are related?
[#2779 ](https://github.com/coredns/coredns/issues/2779)
### 3. Which documentation changes (if any) need to be made?
None.
### 4. Does this introduce a backward incompatible change or deprecation?
No.

### CODE REVIEW

**Comment by miekg** [comment]
fixing this would be swell. Why does this need big.Int? This is just some bit shifting?

**Comment by zamog** [comment]
> fixing this would be swell. Why does this need big.Int? This is just some bit shifting?
For no good reason, updated to use bitwise - commit 25a3af8

**Comment by miekg** [comment]
btw isn't there a neat go package that just does this?

**Comment by zamog** [comment]
> btw isn't there a neat go package that just does this?

didn't find one that has exactly the needed functionality. there are some that implement full cidr calculator like:
https://github.com/apparentlymart/go-cidr - use big int
https://github.com/dspinhirne/netaddr-go - ipv4/ipv6 calculations
but i think it's an overkill to import them for this limited functionality

**Comment by miekg** [comment]
fair enough, let me review

File: plugin/kubernetes/setup.go [hunk 1]
```diff
@@ -121,7 +121,17 @@ func ParseStanza(c *caddy.Controller) (*Kubernetes, error) {
...
+		for i := 0; i < len(zones); i++ {
+			tmp, err := plugin.ClassfulFromCIDR(zones[i])
+			if err == nil {
+				classfulZones = append(classfulZones, tmp...)
+
+			} else {
+				classfulZones = append(classfulZones, zones[i])
+			}
+		}
+		k8s.Zones = classfulZones
```
Comment (miekg) [inline plugin/kubernetes/setup.go:134]: if this requires plugin changes it's a no-go. If we can't do it on the server level than the whole 'reverse-gen' feature will be removed. As in: nice idea, didn't work too bad

File: plugin/normalize.go [hunk 2]
```diff
@@ -141,3 +142,109 @@ func SplitHostPort(s string) (host, port string, ipnet *net.IPNet, err error) {
...
+			cidrs = append(cidrs, networks[i].String())
+		}
+	}
+	return cidrs, err
+}
+
+// NextSubnet returns the next available subnet of the desired mask size
+// starting for the maximum IP of the offset subnet
+// If the IP exceeds the maxium IP then the second return value is true
+func NextSubnet(network *net.IPNet, prefixLen int) (*net.IPNet, bool) {
```
Comment (miekg) [inline plugin/normalize.go:186]: looks copied from https://github.com/apparentlymart/go-cidr/blob/master/cidr/cidr.go#L188

Comment (zamog) [reply plugin/normalize.go:186]: Yes (added a note in the first commit #4501 11a688de5957b505c3b1d74eb09c8cd128ba4390) - seems like it drops after the change

File: plugin/kubernetes/setup.go (hunk 1 above)
Comment (zamog) [reply plugin/kubernetes/setup.go:134]: No problem to implement it on the server

**Comment by miekg** [comment]
See #4538

**Pull request closed without merging**